### PR TITLE
Add a test for Hexbin Linear

### PR DESCRIPTION
**Pull request by lucasricci**
## PR Summary
This PR is to fix the issue #21165:
- A new test were add in ``matplotlib/lib/matplotlib/tests``, with name ``test_hexbin_linear``;
- Also added a image file into ``matplotlib/lib/matplotlib/tests/baseline_images/text_axes``, this image has a use, it is the image that the test will compare with.
- Do it by adding the following:
```python
image_comparison(baseline_images=['hexbin_linear'],
                  extensions=['png'], style='mpl20')
def test_hexbin_linear():

    np.random.seed(19680801)
    n = 100000
    x = np.random.standard_normal(n)
    y = 2.0 + 3.0 * x + 4.0 * np.random.standard_normal(n)

    fig, ax = plt.subplots()
    h = ax.hexbin(x, y, marginals=True, reduce_C_function=np.sum)
    plt.colorbar(h)
```

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by jklymak** [comment]
The new test is failing unfortunately.  Please make sure you test with up-to-date matplotlib main on your machine first.

**Comment by gammamathematics** [comment]
> The new test is failing unfortunately.  Please make sure you test with up-to-date matplotlib main on your machine first.  

Yeah, sure I were going to do that.

**Comment by gammamathematics** [comment]
I'll in approximately 7 hour

**Comment by gammamathematics** [comment]
@jklymak hey, I the ``codecov/project/tests`` test are failing?

**Comment by jklymak** [comment]
New contributors don't get all tests run automatically, but that makes codecov fail (because some tests are not being run). Nothing to worry about.  I manually started your remaining tests.  Thanks!

**Comment by gammamathematics** [comment]
my pleasure, thank you by your help!

**Comment by gammamathematics** [comment]
@jklymak The PR is ready, it passed all tests I open it again.

File: lib/matplotlib/tests/test_axes.py
Comment (QuLogic) [inline lib/matplotlib/tests/test_axes.py:791]: Don't reference a number; describe what this does, or don't include the comment at all.

File: lib/matplotlib/tests/test_axes.py
Comment (QuLogic) [inline lib/matplotlib/tests/test_axes.py:794]: This is a completely new image; you don't need this line.

Comment (gammamathematics) [reply lib/matplotlib/tests/test_axes.py:794]: ok, I will solve it, thanks

File: lib/matplotlib/tests/test_axes.py
Comment (gammamathematics) [reply lib/matplotlib/tests/test_axes.py:791]: ok, I will solve it, thanks

File: lib/matplotlib/tests/test_axes.py
Comment (gammamathematics) [reply lib/matplotlib/tests/test_axes.py:794]: @QuLogic But when I remove it I receive this error message:
```py
===================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.0, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /home/lucas/matplotlib, configfile: pytest.ini
collected 1 item                                                                                                                                                                              

test_axes.py F                                                                                                                                                                          [100%]

========================================================================================== FAILURES ===========================================================================================
___________________________________________________________________________________ test_hexbin_linear[png] ___________________________________________________________________________________

args = (), kwds = {'extension': 'png', 'request': <FixtureRequest for <Function test_hexbin_linear[png]>>}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           matplotlib.testing.exceptions.ImageComparisonFailure: images not close (RMS 0.101):
E           	result_images/test_axes/hexbin_linear.png
E           	result_images/test_axes/hexbin_linear-expected.png
E           	result_images/test_axes/hexbin_linear-failed-diff.png

/home/lucas/.asdf/installs/python/3.10.0/lib/python3.10/contextlib.py:79: ImageComparisonFailure
=================================================================================== short test summary info ===================================================================================
FAILED test_axes.py::test_hexbin_linear[png] - matplotlib.testing.exceptions.ImageComparisonFailure: images not close (RMS 0.101):
====================================================================================== 1 failed in 1.04s ======================================================================================
```

Comment (gammamathematics) [reply lib/matplotlib/tests/test_axes.py:794]: nm, I think is a problem in my PC

Comment (gammamathematics) [reply lib/matplotlib/tests/test_axes.py]: @QuLogic  but when I delete it I receive a error

**Comment by lucasricci** [comment]
Now why this one test are failing?

**Comment by jklymak** [comment]
If you click through the failing test you can see that it was failing to connect to a server to upload artifacts, which happens sometimes, but is probably not this PRs fault.

**Review by jklymak — APPROVED** [review]

**Comment by lucasricci** [comment]
That might be

**Comment by lucasricci** [comment]
@jklymak Can I left like that? With this test failed?

**Comment by jklymak** [comment]
The tests passed, it was just an upload of artifacts that failed, so I would say its fine.

**Comment by lucasricci** [comment]
Ok

**Comment by lucasricci** [comment]
@QuLogic I have already solved the issues you pointed out, thanks!

**Comment by lucasricci** [comment]
I just merged this branch with ``main``, and the tests, that previously succeed in passing, now it is getting me these errors

**Comment by QuLogic** [comment]
If possible, it'd be preferred if you rebased.

**Comment by lucasricci** [comment]
ok, if I should do anything just tag me

**Comment by lucasricci** [comment]
@QuLogic I rebased it, and I think it worked, the tests are now passing,  but ``codecov/project/tests`` no, because as a new contributor, my tests did not run automatically

**Comment by jklymak** [comment]
Your PR has a bunch of extraneous commits, hence the PR cleanliness failure.  I'm not sure if a squash merge will get rid of these, if I was fixing this I would do a rebase, drop the extra commits, and then rebase onto latest master.  Back up your branch before you do this though so you don't lose anything!  Maybe others will have better advice.  

(back up means do `git checkout fix-21165; git checkout -b tmp-backup; git checkout fix-21165` and then do the rebasing.  If you don't know how to rebase, you will probably need to read a guide online, but `git rebase i 12345678` where `12345678` is the last commit in `git log` that you did _not_ author just before the first commit you did author.

**Comment by lucasricci** [comment]
@jklymak I am not seeing any commit that I did not do in ``git log``

**Comment by lucasricci** [comment]
@jklymak Apparently solved

**Comment by lucasricci** [comment]
How can I fix the fact that my baseline image is duplicated? I delete it in my local repo and push it? Any idea?

**Comment by QuLogic** [comment]
You almost had the rebase, but after rebasing, you have to force push up to GitHub. If you pull, you just get back the old commits again and merge them into the history again.

**Comment by QuLogic** [comment]
So I see that #21339 also exists, but has not been updated in some time. However, it was also blocked on #21349, so I think cannot merge this until that is fixed; is that right @timhoffm?

**Comment by lucasricci** [comment]
@QuLogic So I will have to do the rebase again? If so, could you write a small guide please? I've never done a rebase and the texts I found on internet is a little confusing.

**Comment by jklymak** [comment]
You did it right the first time, you just then undid your work by doing a `git pull` before force-pushing your changes to GitHub.  (Also, please don't ask us to write unique guides for every PR... we have our own guide, listed in the docs, and there are lots of docs online.  If you have a specific question about a step, feel free to ask on gitter, thanks for your understanding).

**Comment by lucasricci** [comment]
Ok, Sorry by the request, I am learning a lot contributing with open source software.

**Comment by lucasricci** [comment]
The tests passed now! Great.

**Comment by timhoffm** [comment]
> So I see that #21339 also exists, but has not been updated in some time. However, it was also blocked on #21349, so I think cannot merge this until that is fixed; is that right @timhoffm?

@QuLogic a bit confused by *this* and *that*, but yes, it is preferable if we can solve #21349 before we merge this pull request here.

**Comment by lucasricci** [comment]
@timhoffm So what is gonna be the approach in that case?

**Comment by jklymak** [comment]
Right, this PR is fine, but the marginals may change under #21339, so adding the image, and then re-adding it is unecessary churn in our repo.  So I'll mark this as depending on another PR and move to draft for now.  Thanks for your patience @lucasricci

**Comment by lucasricci** [comment]
@jklymak Just for further information, what is the max period that a PR should be inactivity to be closed?

**Comment by jklymak** [comment]
Just subscribe to upstream issue and when it is merged you can move this one to "ready"

**Comment by lucasricci** [comment]
@jklymak sorry for the ping. What should I do? The other PR are inactive since October, 20th 2021.

**Comment by lucasricci** [comment]
@QuLogic I guess so. You can close the PR

**Comment by lucasricci** [comment]
@QuLogic Idk, but if so, I'll close this PR

**Comment by timhoffm** [comment]
The test is already covered in #21339 now. @lucasricci I'm sorry that there has been overlapping work, but the objective of #21339 changed after we realized that the behavior was in fact correct and a test was then added for this.

Anyway, thank you for the PR. Even though this was not merged, your contributions is valuable to bring the discussion forward.

**Pull request closed without merging**
